git_utils_shared/src/lib.rs

```diff
@@ -0,0 +1,120 @@
+use std::process::Command;
+
+// fn run_command(command: &str, args: [&str]) -> () {}
+
+/// Get the merge base between the two provided branches
+pub fn get_merge_base(branch1: &str, branch2: &str) -> Option<String> {
+    let merge_base_vec = Command::new("git")
+        .arg("merge-base")
+        .arg(branch1)
+        .arg(branch2)
+        .output()
+        .expect("Failed to run git merge-base")
+        .stdout;
+
+    let merge_base = String::from_utf8(merge_base_vec).unwrap();
+    let merge_base = merge_base.trim().to_owned();
+
+    if merge_base.trim().is_empty() {
+        return None;
+    } else {
+        return Some(merge_base);
+    }
+}
+
+/// Show commits on the second branch that are not on the first branch
+pub fn show_uncommon_commit_from_other_branch(branch: &str, other_branch: &str) {
+    let _cmd1 = Command::new("git")
+        .arg("--no-pager")
+        .arg("log")
+        .arg("--pretty=format:%Cgreen%h%Creset %Cred(%an)%Creset [%ad] %Cblue%s%Creset ")
+        .arg(branch)
+        .arg(format!("^{}", other_branch))
+        .status()
+        .expect("Failed to retrieve unique commits");
+    println!("\n");
+}
+
+/// Show the common commit
+pub fn show_common_commit(merge_base: &str) {
+    let _cmd3 = Command::new("git")
+        .arg("--no-pager")
+        .arg("log")
+        .arg("-1")
+        .arg(&merge_base)
+        .status()
+        .expect("Failed to get first common commit");
+}
+
+pub fn get_files_with_word(search: &str, path: &str) -> Option<Vec<String>> {
+    let cmd = Command::new("git")
+        .args([
+            "--no-pager",
+            "grep",
+            "--full-name",
+            "--name-only",
+            search,
+            path,
+        ])
+        .output()
+        .expect("Failed")
+        .stdout;
+    let stdout = String::from_utf8(cmd).unwrap();
+    if stdout.is_empty() {
+        return None;
+    }
+
+    Some(
+        stdout
+            .trim()
+            .split("\n")
+            .map(|path| path.to_owned())
+            .collect::<Vec<String>>(),
+    )
+}
```

Comment:
There is only the need for a single function for finding paths with content, and it should use `git grep` instead of plain grep, and be named accordingly.  The function should accept:

`search` (string)
`paths` (vec of paths)

See notes on where this is invoked.

Specifically, the command should be:

`git grep --files-with-matches --null -E -e "regex" -- <path> <path> <path>`

- `-E` turns on extended regex support
- `-e` is the switch to pass the regex and is required if regexs start with `-`
- `--files-with-matches` only prints the file names
- `-z` null terminates the filenames for better parsing of the output.

```bash
jason@rise21:~/AppCove/ACP7$ git grep --files-with-matches --null -E -e "regex" -- Direct/
Direct/Base/Bundle/Planter/Server2/FileSystem/etc/php.ini\0
```

Each file name is terminated by a `\0` byte (including the last one)

The result of this function should be an unconditional vec of strings (not an Option).  An empty vec is a suitable response if nothing matches.